src/components/ContractorPayment/EditModal/EditModalPresentation.tsx

```diff
@@ -0,0 +1,152 @@
+import { useState } from 'react'
+import type { ContractorDataStrict } from '../types'
+import { useComponentContext } from '@/contexts/ComponentAdapter/useComponentContext'
+import { Flex, Grid, ActionsLayout } from '@/components/Common'
+import { formatNumberAsCurrency } from '@/helpers/formattedStrings'
+import { useLocale } from '@/contexts/LocaleProvider/useLocale'
+
+type ContractorData = ContractorDataStrict
+
+interface ContractorPaymentEditModalProps {
+  contractor: ContractorData
+  onSave: (contractor: ContractorData) => void
+  onCancel: () => void
+}
+
+export const ContractorPaymentEditModal = ({
```

Comment:
Wondering if we update this naming? Seems like this is actually just being used as a screen not a modal? Flows may do some of these as modals but I don't think we need to follow that